### PR TITLE
Fix compatibility with Node.js master

### PR DESCRIPTION
**Pull request by lpinca**
Like #309 but takes into account Node.js versions below 4.

### CODE REVIEW

File: lib/spdy/server.js [hunk 1]
```diff
@@ -189,6 +189,8 @@ proto._onStream = function _onStream (stream) {
     socket = new net.Socket(socketOptions)
   }
 
+  if (process.versions.modules >= 46) { socket.server = this }
```
Comment (lpinca) [inline lib/spdy/server.js]: Currently this is only needed for Node.js master but also handles the unlikely case where https://github.com/nodejs/node/pull/11926 will be backported to Node.js 6 and 4 (`modules >= 46`).

Comment (ronkorving) [inline lib/spdy/server.js]: This kind of code would be well served with a code comment :)

Comment (lpinca) [inline lib/spdy/server.js]: Agreed will add in a bit.

**Comment by lpinca** [comment]
@diasdavid can you please take a look when you have time?
https://github.com/nodejs/node/pull/11926 has not yet landed in a v7 release as it breaks this module.
It would be great to know if this fix/workaround is acceptable.

**Comment by refack** [comment]
Compatibility restoration confirmed
Ref: https://github.com/nodejs/node/wiki/CITGM-Status

**Pull request merged**